torch/_inductor/codegen/cpp.py

```diff
@@ -3235,6 +3178,116 @@ def select_tiling(
+                                            _node.target, negative_op_type_count
+                                        )
+                            elif _node.target == "load":
+                                index = sub_block.body.indexing_exprs[
+                                    _node.args[-1].args[0]
+                                ]
+                                if len(tiling_indices) == 1:
+                                    stride = _try_get_stride(
+                                        index, tiling_factor, tiling_indices
+                                    )
+                                    if stride is not None and stride not in [0, 1]:
+                                        _update_negative_op_count(
+                                            _node.target, negative_op_type_count
+                                        )
+                            elif _node.target == "store":
+                                index = sub_block.body.indexing_exprs[
+                                    _node.args[2].args[0]
+                                ]
+                                if len(tiling_indices) == 1:
+                                    stride = _try_get_stride(
+                                        index, tiling_factor, tiling_indices
+                                    )
+                                    if stride is not None and stride != 1:
+                                        _update_negative_op_count(
+                                            _node.target, negative_op_type_count
+                                        )
+
+                            if isinstance(_node.target, str) and not (
+                                _node.target.startswith("masked_subblock")
+                                or _node.target
+                                in ["ops", "output", "constant", "get_index"]
+                            ):
+                                if _node.target not in op_type_count:
+                                    op_type_count[_node.target] = 1
+                                else:
+                                    op_type_count[_node.target] += 1
+
+                total_op_count = sum(op_type_count.values())
+                negative_op_count = sum(negative_op_type_count.values())
+                threshold = 0.03
+                if (
+                    total_op_count > 0
+                    and negative_op_count / total_op_count >= threshold
+                ):
+                    # Too many non-contiguous load/store/index_expr which hurts the
+                    # vectorization performance. Disable vectorization when exceeding
+                    # the threshold.
+                    return [], []
+
+                if (
+                    not reduction_group
+                    and group
+                    and not has_free_symbols(group[-1])
+                    and group[-1] < tiling_factor / 2
```

Comment:
Shouldn't hard-code `-1` here?

Reply:
Thanks for the comment, since we only meet regression of tile1D case, change -1 to `tiling_indices[0]`.